p2panda-net-next/src/actors/iroh/endpoint.rs

```diff
@@ -186,6 +195,17 @@ impl ThreadLocalActor for IrohEndpoint {
                     .bind()
                     .await?;
 
+                // Listen for changes to our endpoint's addresses.
+                let watch_addr_handle = {
+                    let endpoint = endpoint.clone();
+                    let myself = myself.clone();
+                    tokio::spawn(async move {
+                        while let Ok(addr) = endpoint.watch_addr().updated().await {
+                            let _ = myself.send_message(ToIrohEndpoint::AddressUpdate(addr.addrs));
+                        }
+                    })
+                };
```

Comment:
This logic is already handled through the node info watchers. We receive this information the address book discovery service ("resolve" and "publish") and can watch our own address to receive this data.

Designing it like this allows other services to just look at the address book without knowing about the endpoint specifically.